#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: This dpatch addes many locales especially for UTF-8 and 
# DP:		ISO-8859-15, to display debconf locale menu using
# DP:		SUPPORTED.orig. This dpatch is based on RedHat cygnus-to-
# DP:		redhat patch.
# DP: Author: GOTO Masanori <gotom@debian.org>
# DP: Upstream status: Debian/RedHat-Specific
# DP: Status Details: 
# DP: Date: 2003-02-09

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.
--- localedata/SUPPORTED.org	2003-02-17 11:07:03.000000000 +0900
+++ localedata/SUPPORTED	2003-02-17 11:13:32.000000000 +0900
@@ -30,6 +30,7 @@
 cs_CZ/ISO-8859-2 \
 cy_GB/ISO-8859-14 \
 da_DK/ISO-8859-1 \
+da_DK.ISO-8859-15/ISO-8859-15 \
 de_AT/ISO-8859-1 \
 de_AT@euro/ISO-8859-15 \
 de_BE/ISO-8859-1 \
@@ -48,6 +49,7 @@
 en_DK/ISO-8859-1 \
 en_GB/ISO-8859-1 \
 en_GB.UTF-8/UTF-8 \
+en_GB.ISO-8859-15/ISO-8859-15 \
 en_HK/ISO-8859-1 \
 en_IE/ISO-8859-1 \
 en_IE@euro/ISO-8859-15 \
@@ -57,6 +59,7 @@
 en_SG/ISO-8859-1 \
 en_US/ISO-8859-1 \
 en_US.UTF-8/UTF-8 \
+en_US.ISO-8859-15/ISO-8859-15 \
 en_ZA/ISO-8859-1 \
 en_ZW/ISO-8859-1 \
 es_AR/ISO-8859-1 \
@@ -150,6 +153,7 @@
 sv_FI/ISO-8859-1 \
 sv_FI@euro/ISO-8859-15 \
 sv_SE/ISO-8859-1 \
+sv_SE.ISO-8859-15/ISO-8859-15 \
 ta_IN/UTF-8 \
 te_IN/UTF-8 \
 tg_TJ/KOI8-T \
@@ -173,3 +177,124 @@
 zh_TW/BIG5 \
 zh_TW.EUC-TW/EUC-TW \
 zh_TW.UTF-8/UTF-8 \
+ar_AE.UTF-8/UTF-8 \
+ar_BH.UTF-8/UTF-8 \
+ar_DZ.UTF-8/UTF-8 \
+ar_EG.UTF-8/UTF-8 \
+ar_IQ.UTF-8/UTF-8 \
+ar_JO.UTF-8/UTF-8 \
+ar_KW.UTF-8/UTF-8 \
+ar_LB.UTF-8/UTF-8 \
+ar_LY.UTF-8/UTF-8 \
+ar_MA.UTF-8/UTF-8 \
+ar_OM.UTF-8/UTF-8 \
+ar_QA.UTF-8/UTF-8 \
+ar_SA.UTF-8/UTF-8 \
+ar_SD.UTF-8/UTF-8 \
+ar_SY.UTF-8/UTF-8 \
+ar_TN.UTF-8/UTF-8 \
+ar_YE.UTF-8/UTF-8 \
+be_BY.UTF-8/UTF-8 \
+bg_BG.UTF-8/UTF-8 \
+ca_ES.UTF-8/UTF-8 \
+ca_ES.UTF-8@euro/UTF-8 \
+cs_CZ.UTF-8/UTF-8 \
+da_DK.UTF-8/UTF-8 \
+de_AT.UTF-8/UTF-8 \
+de_AT.UTF-8@euro/UTF-8 \
+de_BE.UTF-8/UTF-8 \
+de_BE.UTF-8@euro/UTF-8 \
+de_CH.UTF-8/UTF-8 \
+de_DE.UTF-8@euro/UTF-8 \
+de_LU.UTF-8/UTF-8 \
+de_LU.UTF-8@euro/UTF-8 \
+en_AU.UTF-8/UTF-8 \
+en_BW.UTF-8/UTF-8 \
+en_CA.UTF-8/UTF-8 \
+en_DK.UTF-8/UTF-8 \
+en_HK.UTF-8/UTF-8 \
+en_IE.UTF-8/UTF-8 \
+en_IE.UTF-8@euro/UTF-8 \
+en_NZ.UTF-8/UTF-8 \
+en_PH.UTF-8/UTF-8 \
+en_SG.UTF-8/UTF-8 \
+en_ZA.UTF-8/UTF-8 \
+en_ZW.UTF-8/UTF-8 \
+es_AR.UTF-8/UTF-8 \
+es_BO.UTF-8/UTF-8 \
+es_CL.UTF-8/UTF-8 \
+es_CO.UTF-8/UTF-8 \
+es_CR.UTF-8/UTF-8 \
+es_DO.UTF-8/UTF-8 \
+es_EC.UTF-8/UTF-8 \
+es_ES.UTF-8/UTF-8 \
+es_ES.UTF-8@euro/UTF-8 \
+es_GT.UTF-8/UTF-8 \
+es_HN.UTF-8/UTF-8 \
+es_MX.UTF-8/UTF-8 \
+es_NI.UTF-8/UTF-8 \
+es_PA.UTF-8/UTF-8 \
+es_PE.UTF-8/UTF-8 \
+es_PR.UTF-8/UTF-8 \
+es_PY.UTF-8/UTF-8 \
+es_SV.UTF-8/UTF-8 \
+es_US.UTF-8/UTF-8 \
+es_UY.UTF-8/UTF-8 \
+es_VE.UTF-8/UTF-8 \
+et_EE.UTF-8/UTF-8 \
+eu_ES.UTF-8/UTF-8 \
+eu_ES.UTF-8@euro/UTF-8 \
+fi_FI.UTF-8/UTF-8 \
+fi_FI.UTF-8@euro/UTF-8 \
+fo_FO.UTF-8/UTF-8 \
+fr_BE.UTF-8/UTF-8 \
+fr_BE.UTF-8@euro/UTF-8 \
+fr_CA.UTF-8/UTF-8 \
+fr_CH.UTF-8/UTF-8 \
+fr_FR.UTF-8@euro/UTF-8 \
+fr_LU.UTF-8/UTF-8 \
+fr_LU.UTF-8@euro/UTF-8 \
+ga_IE.UTF-8/UTF-8 \
+ga_IE.UTF-8@euro/UTF-8 \
+gl_ES.UTF-8/UTF-8 \
+gl_ES.UTF-8@euro/UTF-8 \
+gv_GB.UTF-8/UTF-8 \
+he_IL.UTF-8/UTF-8 \
+hr_HR.UTF-8/UTF-8 \
+hu_HU.UTF-8/UTF-8 \
+id_ID.UTF-8/UTF-8 \
+is_IS.UTF-8/UTF-8 \
+it_CH.UTF-8/UTF-8 \
+it_IT.UTF-8/UTF-8 \
+it_IT.UTF-8@euro/UTF-8 \
+iw_IL.UTF-8/UTF-8 \
+kl_GL.UTF-8/UTF-8 \
+kw_GB.UTF-8/UTF-8 \
+lt_LT.UTF-8/UTF-8 \
+lv_LV.UTF-8/UTF-8 \
+mk_MK.UTF-8/UTF-8 \
+ms_MY.UTF-8/UTF-8 \
+mt_MT.UTF-8/UTF-8 \
+nl_BE.UTF-8/UTF-8 \
+nl_BE.UTF-8@euro/UTF-8 \
+nl_NL.UTF-8/UTF-8 \
+nl_NL.UTF-8@euro/UTF-8 \
+nn_NO.UTF-8/UTF-8 \
+no_NO.UTF-8/UTF-8 \
+pl_PL.UTF-8/UTF-8 \
+pt_BR.UTF-8/UTF-8 \
+pt_PT.UTF-8/UTF-8 \
+pt_PT.UTF-8@euro/UTF-8 \
+ro_RO.UTF-8/UTF-8 \
+ru_UA.UTF-8/UTF-8 \
+sk_SK.UTF-8/UTF-8 \
+sl_SI.UTF-8/UTF-8 \
+sq_AL.UTF-8/UTF-8 \
+sr_YU.UTF-8/UTF-8 \
+sr_YU.UTF-8@cyrillic/UTF-8 \
+sv_FI.UTF-8/UTF-8 \
+sv_FI.UTF-8@euro/UTF-8 \
+sv_SE.UTF-8/UTF-8 \
+th_TH.UTF-8/UTF-8 \
+tr_TR.UTF-8/UTF-8 \
+uk_UA.UTF-8/UTF-8 \
